### DISCUSSION TITLE
Managing Quartz in WebApplicationFactory tests

### DISCUSSION THREAD

**Discussion by drusellers** [quartznet/quartznet#2277] [Q&A] [answered]
is there a recipe for managing quartz in a "non-run" state? I'm writing tests with `WebApplicationFactory` and I'm trying to reduce test duration times. I'd like to have quartz added to the container so that I can see that jobs got added etc. This means I need the migrations to run. But I don't want the jobs to run, at all. 

When I was looking at dotTrace I was seeing a good amount of time (~12sec?) being spent in `QuartzSchedulerThread.Run` - since I don't think I need Quartz to run - I'm wondering if that can be turned off in unit test land. 

Tried `UseZeroSizeThreadPool`

also tried

```csharp
services.AddQuartz(q =>
{
    var maxConcurrency = 0;
    q.MisfireThreshold = TimeSpan.FromSeconds(60);
    q.UseDefaultThreadPool(options => { options.MaxConcurrency = maxConcurrency; });
    q.SetProperty(StdSchedulerFactory.PropertySchedulerMaxBatchSize, maxConcurrency.ToString());
});
```

any ideas on how to best speed up running these kinds of tests?

**Comment by lahma** [comment, marked as answer]
You could try to remove the hosted service in `WebApplicationFactory` context: https://stackoverflow.com/questions/56341035/how-to-disable-services-with-webapplicationfactory
